scene/resources/shader.cpp

```diff
@@ -76,8 +76,9 @@ void Shader::get_param_list(List<PropertyInfo> *p_params) const {
 		if (default_textures.has(pi.name)) { //do not show default textures
 			continue;
 		}
+		String original_name = pi.name;
 		pi.name = "shader_param/" + pi.name;
-		params_cache[pi.name] = pi.name;
+		params_cache[pi.name] = original_name;
```

Comment:
I'm not sure this fix restores the original behavior of the code prior to #50511.

It used to be:
```cpp
for (List<PropertyInfo>::Element *E = local.front(); E; E = E->next()) {
		PropertyInfo pi = E->get();
		pi.name = "shader_param/" + pi.name;
		params_cache[pi.name] = E->get().name;
```
i.e. `pi` was a copy which got its named modified.

With the new code, `pi` is a reference, so we're modifying the name of `local`'s copy of the param list IINM.

I don't know if it's a problem, but that's worth noting.

Reply:
Yep, sorry about the quick merge - but I've really quite tired of this bug, and this PR really fixes it.

Reply:
@akien-mga in either case ``pi`` is modified before it is stored in p_params. So that is likely not an issue. The real issue is that the copy of the Property info stored in local  (e.g. the copy returned by ``RenderingServer::get_singleton()->shader_get_param_list(shader, &local);``) will be incorrectly changed. 

Although, now that I think about it. I remember JFons talking about this new iterator approach creating copies unless we added const or something. Perhaps this iterator is making a copy of ``pi`` anyway?